internal/rest/client/client.go

```diff
@@ -281,10 +281,15 @@ func (c *Client) QueryStruct(ctx context.Context, method string, endpointType ty
 		localURL = &newURL
 	}
 
+	pathPrefix := "/" + string(endpointType)
+	if endpointType == "" {
+		pathPrefix = ""
+	}
+
 	localURL.URL.Host = c.url.URL.Host
 	localURL.URL.Scheme = c.url.URL.Scheme
-	localURL.URL.Path = "/" + string(endpointType) + localURL.URL.Path
-	localURL.URL.RawPath = "/" + string(endpointType) + localURL.URL.RawPath
+	localURL.URL.Path = pathPrefix + localURL.URL.Path
```

Comment:
We can just do `filepath.Join("/", string(endpointType), localURL.URL.Path)` as it will skip path elements which are empty.

Reply:
No need for the extra `if`.

Reply:
Neat! Thanks will update :slightly_smiling_face: